external/CMakeLists.txt

```diff
@@ -50,14 +50,20 @@ add_library(ngtcp2::crypto
     ngtcp2_crypto)  # ngtcp2 crypto (and implicitly base ngtcp2)
 
 # oxen-logging
-if(BUILD_STATIC_DEPS)
-    set(OXEN_LOGGING_FORCE_SUBMODULES ON CACHE INTERNAL "")
+if (NOT TARGET oxen::logging)
+    if(BUILD_STATIC_DEPS)
+        set(OXEN_LOGGING_FORCE_SUBMODULES ON CACHE INTERNAL "")
+    endif()
+    set(OXEN_LOGGING_SOURCE_ROOT "${PROJECT_SOURCE_DIR}" CACHE INTERNAL "")
+    add_subdirectory(oxen-logging)
+else()
+    set(OXEN_LOGGING_SOURCE_ROOT "${OXEN_LOGGING_SOURCE_ROOT};${PROJECT_SOURCE_DIR}" CACHE INTERNAL "")
```

Comment:
Need to investigate whether this appends multiple times if you re-run cmake multiple times from an existing build directory.  (The final value will end up in the build dir's CMakeCache.txt).